### DISCUSSION TITLE
Any way of getting record id before creating

### DISCUSSION THREAD

**Discussion by nikhil-varma-mudunuri** [pocketbase/pocketbase#3407] [Q&A] [answered]
my requirement is I have to create and record, in that record, i have to include the record id in one column, (my solution : is to give custom id, is there any other way?)

**Comment by ganigeorgiev** [comment, marked as answer]
If the record was submitted without an id, then you have to manually set it if you want to access it.

You can generate it with the `$security` helpers as mentioned in https://github.com/pocketbase/pocketbase/discussions/3408, OR use the `e.record.refreshId()` method.
